nulib/meadow
# CSV Updates: Add better validation / tolerance for query rows & subject roles

# Summary 
Prevent CSV metadata updates from getting stuck during validation

# Specific Changes in this PR
- Find real header row to start from without being rigid about “1 query row, 1 header row, N content rows”
- Report an error when a valid header row can't be found
- Check to make sure subject roles are valid
- Error out a job that's already been reset/retried 3 times
- Move/rename text fixture CSVs for readability

# Version bump required by the PR

See [Semantic Versioning 2.0.0](https://semver.org/) for help discerning which is required.


# Steps to Test

- Send through a good metadata update and make sure it works
- Send through a CSV update sheet with no query row and make sure it works
- Send through a CSV update sheet with no header row or the wrong number of headers and make sure it works
- Send through a CSV update sheet with too many rows before the headers and make sure it works
- Send through a CSV update sheet with a bad subject role and make sure it fails validation with a coded term error

# :rocket: Deployment Notes

**Note** - __if you check any of these boxes go to the [(always open) `main` <- `staging` PR](https://github.com/nulib/meadow/pulls) and add detailed notes and instructions to help out others who may end up deploying your changes to production__

- Backward compatible API changes
- Backwards-incompatible API changes


# Tested/Verified
